File: baekjoon/10828.py
import sys

li = []
re = int(input())
n = 0
for i in range(re) :
    a = sys.stdin.readline().split()
    if a[0] == "push" :
        li.append(a[1])
        n += 1
    elif a[0] == "pop" :
        if n != 0 :
            n -= 1
            print(li.pop())
        else :
            print("-1")
    elif a[0] == "size" :
        print(n)
    elif a[0] == "empty" :
       if n == 0 : print("1")
       elif n != 0 : print("0")
    elif a[0] == "top" :
        if n == 0 : print("-1")
        else : print(li[n-1])

File: baekjoon/10828.cpp
#include<iostream>
#include<stack>

using namespace std;

int main(void){
    stack<int> s;

    int n = 0;
    scanf("%d",&n);
    string in;
    int target = 0;
    for(int i = 0; i < n; i++){
        cin >> in;
        if  (in == "push") {
            int n = 0;
            scanf("%d", &n);
            s.push(n);
        }
        else if (in == "pop") {
            if (s.empty()){
                printf("%d\n", -1);
            }
            else {
                printf("%d\n", s.top());
                s.pop();
            }
        }
        else if (in == "size") printf("%lu\n", s.size());
        else if (in == "empty") printf("%d\n", s.empty());
        else if (in == "top"){
            if (s.empty()){
                printf("%d\n", -1);
            }
            else {
                printf("%d\n", s.top());
            }
        } 
    }
}